### ISSUE TITLE
Player 'Default' has stopped responding! displayed but still working Ok

### ISSUE THREAD

**Issue by Eribotes** [fatg3erman/RompR#157]
I recently noticed that I'm inconsistently getting a "Player 'Default' has stopped responding!" persistent warning drop-down displayed at the top of of the desktop UI.  However, Rompr still seems to be working correctly (with the exception of the volume control, which has never worked for me due, I believe, to an issue unrelated to rompr).  Refreshing the browser (Debian Firefox ESR) displays a temporary drop-down saying that it has connected to the default player (mpd) ok.  The persistent "Player 'Default' has stopped reponding!" drop-down may reappear after a minute or two, or it may not, seemingly at random - I can't link it to anything.

I am running V2.10 with mpd on a RPi2 fitted with an IQ Audio DAC Pro.  The OS is Raspberry Pi OS Bookworm (old stable).

I don't know exactly when this issue started but noticed it while monitoring rompr after I had removed Jack from the audio chain because I couldn't get it to do what I wanted properly. Until I removed Jack the audio chain was rompr -> mpd -> Jack -> Alsa -> h/w.  After removing Jack the chain is now rompr -> mpd -> Alsa -> h/w.

I do now have an issue with mpd not being able to find the PCM mixer in the IQ Audio DAC Pro, despite Alsa seeming to identify it and all of its controls correctly.  I suspect this is the reason why rompr's volume control does not work for me.

It is also possible that it is the IQ Audio DAC Pro PCM issue that is prompting rompr to display the persistent warning and that Jack, when I had it installed, was masking it by sitting between mpd and Alsa.

I have reported this as a bug for two reasons: because the warning message seems to be incorrect; mpd is still responding, and because the warning, being persistent, blocks part of the title of what is being played and can't reliably be cleared.

A preference or setting option to make non-fatal warnings non-persistent would stop such drop-downs from blocking the current title.  A more specific description of problems detected by Rompr would be helpful.

**Comment by fatg3erman** [comment]
That warning is coming from the Websocket Server https://fatg3erman.github.io/RompR/Rompr-And-MPD. It's permanent because a part of the system is not functioning correctly and so it may not perform as expected. I want people to report these things :)

It would seem that your websocket server is either not starting, or is crashing. You could investigate by running it manually from a terminal and seeing what happens. Alternatively, I don't remember offhand, but I think setting 'Websocket' to an empty value in the Player definition will make RompR not attempt to use it. Your choice.

The Alsa mixer not working with MPD is a known bug with mpd when using many of the Pi HATs. My preference is to use Mopidy (alongside mopidy-mpd and mopidy-alsamixer) which does work correctly with these devices, and will also resolve your websocket server issue.

**Comment by Eribotes** [comment]
Updating this to say that thanks to the comments by 'borine' in this MPD thread:

https://github.com/MusicPlayerDaemon/MPD/issues/374

I was able to get the IQaudIODAC mixer control recognised by MPD and RompR, so I can now control the volume via RompR.  The info given in that thread may be of help to users of other pihat DACs.

However, I'm still getting the "Player 'Default' has stopped responding!" drop-down warnings appearing, somewhat inconsistently.  Although I haven't yet been able to identify a clear pattern to their appearance I spotted something in the syslog which is clearly anomalous.

I'm running MPD as a system service, as the 'mpd' user, and 'systemctl status mpd' shows it running ok since the last boot, but in the syslog I can also see subsequent attempts to start MPD, which fail because its socket is already in use, presumably by the already-running system instance.

From futher research it seems that mpd can be started by systemd at a 'user' level if it thinks that MPD is not already running at 'system' level and I think this may be a factor: on one occasion I found an MPD start-up failure in the syslog that seemed to correspond to me ssh'ing into the system.  This wasn't conclusive though, because logging out and then back in again didn't seem to raise the same or any other error.  Also, I generally only ssh in to the (headless) system once per week, for housekeeping, and other MPD start-up failures are occurring between logins.

At first glance, the failed MPD restart attempts do not seem to correspond to the appearances of the RompR drop-down warnings, which seems to occur far more frequently than the number of logged MPD start failures.

I tried looking into your suggestion re the websocket but have only just realised, as I was writing this, that I was looking for it in the MPD conf, it being the 'player', in my mind, whereas now I think I realise you meant the RompR conf - doh!

Sorry for the delay in getting back to this but been busy.
